espflash/src/flasher/mod.rs

```diff
@@ -60,27 +60,38 @@ pub(crate) mod stubs;
 #[repr(u8)]
 pub enum FlashFrequency {
     /// 12 MHz
+    #[serde(rename = "12MHz")]
```

Comment:
I mentioned this in the commit comments, but just to explicitly call it out: these `rename` attributes will break anything that tries to deserialize FlashFrequency/FlashMode/FlashSize values that were serialized by old versions of the code.

However, I didn't see anything that appeared to be using this serialization.  The serialization support was added in #528, for https://github.com/probe-rs/probe-rs/pull/1952, but it seems like that functionality wasn't ultimately used in probe-rs as best as I can tell.

Reply:
@bugadani Will this cause issues in probe-rs?

Reply:
I don't think so. I think what happened here is that #528 was merged, then I relaised I can simplify something and didn't end up needing serde.